### ISSUE TITLE
[HOLD for payment 2023-07-17] [$2000] Feature Request: Update header for 6 person DMs to show all avatars

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15929]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem:
DM members in group chats are capped at a maximum of 4 avatars

## Solution:
DMs are capped at 6 members and then add +1 to show more members, Once we get to 5, we use a smaller avatar size

## Context/Examples/Screenshots/Notes:
<img width="1701" alt="Screenshot 2023-03-13 at 4 50 40 PM" src="https://user-images.githubusercontent.com/43996225/224852401-85b13951-19a0-416c-81d5-05765092f206.png">

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**
**Issue reported by:**@puneetlath
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678715475290659

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c018b3452ff97bbe</li>
        <li>Upwork Job ID: 1635771179154558976</li>
        <li>Last Price Increase: 2023-06-16</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mateocole (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by shawnborton** [comment]
Looks good to me!

**Comment by mateocole** [comment]
@shawnborton is this something we will likely be looking at updating internally?

**Comment by puneetlath** [comment]
This can be done by external contributors.

**Comment by mateocole** [comment]
got it in that case going to mark as `external`

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01c018b3452ff97bbe

**Comment by MelvinBot** [comment]
Current assignee @mateocole is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @NikkiWines (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by allroundexperts** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Avatar header in the report chat is limited to 4 avatars.

### What is the root cause of that problem?
We've set `CONST.REPORT.MAX_PREVIEW_AVATARS` to 4.

### What changes do you think we should make in order to solve the problem?

We can change `CONST.REPORT.MAX_PREVIEW_AVATARS` to `6`. After that, we can add a condition to this file like:

```
    const avatarSize = props.icons.length < 4 ? CONST.AVATAR_SIZE.LARGE : CONST.AVATAR_SIZE.MEDIUM;
````
This will cause the avatar to show in medium size if there are more than 4 of them.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by abdulrahuman5196** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Feature request to show 6 person Avatars in chat

### What is the root cause of that problem?
Currently we are using room header component which is limited to 4 avatars being shown.
It has following disadvantages to have this feature request working
- It is hardcoded to show 4 icons ([CONST.REPORT.MAX_PREVIEW_AVATARS](https://github.com/Expensify/App/blob/main/src/components/RoomHeaderAvatars.js#L39))
- The styles are hardcoded to show large icon sizes. ([CONST.AVATAR_SIZE.LARGE](https://github.com/Expensify/App/blob/main/src/components/RoomHeaderAvatars.js#L55))
- Same goes for all other style factors like '+1' [text style](https://github.com/Expensify/App/blob/main/src/components/RoomHeaderAvatars.js#L74)

### What changes do you think we should make in order to solve the problem?

**Background**
We have already made MultipleAvatars Component generic to support multiple sizes of avatar dynamically as part of this issue - https://github.com/Expensify/App/issues/15083#issuecomment-1441221893, https://github.com/Expensify/App/issues/15083#issuecomment-1455196772
The changes are in PR review state - https://github.com/Expensify/App/pull/15672
This supports
- Dynamically resizing based on the avatar sizes provided like SMALL, LARGE, LARGE_BORDERED, etc.
- Show related border width based on the avatar
- Shows +1 font size based on the avatar size
- And other style changes to make it generic
<img width="400" alt="Screenshot 2023-03-06 at 1 42 25 AM" src="https://user-images.githubusercontent.com/46707890/225222378-5cd9ea3e-f376-4bcb-bec6-73eb35140c7d.png">
<img width="400" alt="Screenshot 2023-03-06 at 1 42 50 AM" src="https://user-images.githubusercontent.com/46707890/225222399-954b350e-d554-4fec-a29f-eae53226ee8e.png">
<img width="400" alt="Screenshot 2023-03-06 at 1 43 23 AM" src="https://user-images.githubusercontent.com/46707890/225222408-70e2566c-5e99-4883-af74-514e29e0e881.png">
<img width="400" alt="Screenshot 2023-03-06 at 1 43 48 AM" src="https://user-images.githubusercontent.com/46707890/225222412-ddccedb7-7d94-4349-aeb4-31ccaa6c6197.png">
<img width="400" alt="Screenshot 2023-03-06 at 1 45 01 AM" src="https://user-images.githubusercontent.com/46707890/225222416-d73f0a66-39e2-43bd-bde5-d00fd671dc7d.png">


**Changes required**
- We should replace the use of RoomHeaderAvatar to MultipleAvatar component in ReportActionItemCreated, this way we can show different avatar sizes based on number icons to show or screen size. For eg. On phones when we have greater than 5/6 users, we can choose to show SMALL avatar and when we less users we can choose to show LARGE avatar size. I am just saying this as example, we can choose any of the avatar sizes for this. Or we can also choose to show one avatar size for all cases as well
- We should expose a param like 'maxAvatarToShow' in MultipleAvatar component for which in this case we can provide 6 so that we will display 6 avatars and show +1 after that if required. We will also have to update logic in MultipleAvatars to support number of avatars to show based on this external param in the MultipleAvatars component
- I can see the width of one avatar overlapping with other is less in this chat page mock compared to the invite page mock, so we can expose a param like 'overlapSize' to provide the size on avatar should overlap on other. And we should add the logic for MultipleAvatars component use the overlapSize from params.(Only if this is required)
The code changes for both the above changes need to be done here - https://github.com/Expensify/App/blob/5cba7218adbc0d4f8383b312beb7b9d90d2f8898/src/components/MultipleAvatars.js#L89-L174 to update the use of dynamic params from components using MultipleAvatars
(I am not adding exact code diffs as its against current guidelines)
Optional: RoomHeaderAvatar is also used on ReportDetailsPage as well, we can replace there as well if required. 


@shawnborton On high level we can re-use the MultipleAvatars component made in a generic way as part of the Invite Message page(https://github.com/Expensify/App/issues/15083) for this issue as well with some modifications as mentioned above 

### What alternative solutions did you explore? (Optional)
NA

**Comment by Snehal-Techforce** [comment]
### Proposal

**Problem:**
DM members in group chats are capped at a maximum of 4 avatars

**Solution:**

1. MAX_PREVIEW_AVATARS constant value needs to change from 4 to 6 in CONST.js file.

![image](https://user-images.githubusercontent.com/123076498/225297183-12923000-d080-4efb-b9d7-528c6ef0aae7.png)

2. CSS change would be required in RoomHeaderAvatars.js file as below.
![image](https://user-images.githubusercontent.com/123076498/225297320-0dcd36c4-b783-452d-9d80-ef21b57e23d6.png)

**Expected Output:**

1. For 6 avatars it will be displayed as below:
![image](https://user-images.githubusercontent.com/123076498/225297634-37e17331-6d66-4e51-ad2a-aabfe7f32fb4.png)

2. For 6+ avatars , it will be displayed as below:
![image](https://user-images.githubusercontent.com/123076498/225297693-8cc958e4-ff9d-4404-9475-7a46982d49a1.png)

3. For <6 avatars , it will be displayed as below:
![image](https://user-images.githubusercontent.com/123076498/225297725-242a7421-2157-4d03-933e-6cbc1ada279d.png)

**Comment by MelvinBot** [comment]
📣 @Snehal-Techforce! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by mananjadhav** [comment]
Going to review proposals in a few hours.

**Comment by Snehal-Techforce** [comment]
Contributor details
Your Expensify account email: reach@techforceglobal.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01fe25025615ae221e?viewMode=1

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by NikkiWines** [comment]
Going to be OOO next week so going to reapply the `External` label so that I don't hold things up!

**Comment by MelvinBot** [comment]
Current assignee @mateocole is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @mananjadhav is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @roryabraham (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mananjadhav** [comment]
Apologies for the delay, I was OOO. I'll check the proposals today. I did go through them once and I am checking if @abdulrahuman5196's proposal of making `MultipleAvatars` work would be beneficial.

@Snehal-Techforce quick feedback, please don't add Code diffs in the proposals. They would ignored from the review, we're focused on explaining the technical details of the proposals, along with some pseudocode may be.

**Comment by Snehal-Techforce** [comment]
> Thank you for your comment and providing feedback as well @mananjadhav. For upcoming proposals, I will not add Code diffs.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by roryabraham** [comment]
I like @abdulrahuman5196's proposal to re-use the `MultipleAvatars` component and get rid of `RoomHeadersAvatar`. I want to add a few additional requirements:

1. Must get rid of other instances of `RoomHeadersAvatar` and standardize on `MultipleAvatars`
1. Should support up to 7 avatars (the maximum for a group chat)
1. Should only use smaller icons if the screen is narrow – on wider screens should continue to use the current avatar size.
1. Storybook stories for the `MultipleAvatars` component

**Comment by MelvinBot** [comment]
📣 @abdulrahuman5196 You have been assigned to this job by @roryabraham!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by mananjadhav** [comment]
I think we should put this on HOLD until https://github.com/Expensify/App/pull/15672 is merged but I liked @abdulrahuman5196 suggestion of using `MultipleAvatars` component.

**Comment by mananjadhav** [comment]
Ohh @roryabraham I just saw your comment, what do you think of putting this on Hold until the linked issue is merged as it is  still WIP and will conflict with the current design. Agreed with the additional requirements though.

**Comment by roryabraham** [comment]
HOLDing on https://github.com/Expensify/App/pull/15672

**Comment by abdulrahuman5196** [comment]
> I like @abdulrahuman5196's proposal to re-use the `MultipleAvatars` component and get rid of `RoomHeadersAvatar`. I want to add a few additional requirements:
> 
>     1. Must get rid of other instances of `RoomHeadersAvatar` and standardize on `MultipleAvatars`
> 
>     2. Should support up to 7 avatars (the maximum for a group chat)
> 
>     3. Should only use smaller icons if the screen is narrow – on wider screens should continue to use the current avatar size.
> 
>     4. Storybook stories for the `MultipleAvatars` component

@roryabraham 
Thank your assigning the issue to me. I agree to hold this issue until the refractor PR is merged. We are on final stage of review there and should be closing out the PR soon. After that I will start working on the requirements here. 

Wanted to confirm on the below meanwhile.
First 2 requests are fine.

>3. Should only use smaller icons if the screen is narrow – on wider screens should continue to use the current avatar size.

On the OP it is mentioned  "Once we get to 5, we use a smaller avatar size", then this should only apply to smallScreenWidth cases right? The UI permutation will be,

case 1 : isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE
case 2: isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.SMALL
case 3: not isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE
case 3: not isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.LARGE
Correct me if any of the above case needs to be changed.


>4. Storybook stories for the `MultipleAvatars` component

I haven't added storybook for a component yet on expensify app. Could you kindly point me to a PR/reference, so that I can do the same for MultipleAvatars component as well.

**Comment by roryabraham** [comment]
@abdulrahuman5196 yep, that looks correct 👍🏼

**Comment by mateocole** [comment]
Going to be OOO, reassigning bug on this

**Comment by roryabraham** [comment]
Moving this to weekly while it's on HOLD

**Comment by slafortune** [comment]
Still on hold for https://github.com/Expensify/App/pull/15672

**Comment by abdulrahuman5196** [comment]
The PR https://github.com/Expensify/App/pull/15672 has been merged.
Just to confirm, before I start working on this feature request since it has been couple of weeks before this issue was put on hold.

1) This is the new invite members screen using the new multiple avatars component. We are expecting the same with the support to show all avatars for big screen and for small screen with over 4 avatars we choose smaller size.
2) Remove RoomHeadersAvatar usages and replace it with MultipleAvatars.
3) Create storybook for multiple avatars
@roryabraham @mananjadhav 

<img width="1197" alt="Screenshot 2023-04-29 at 9 02 58 PM" src="https://user-images.githubusercontent.com/46707890/235311040-37ea9e23-9e6f-4112-852b-2dd5b471fa4b.png">

**Comment by puneetlath** [comment]
Yes, that sounds about right to me. Main thing is that, given that you can have at most 8 people in a group DM, we should make it so that all the avatars fit in the room header in all scenarios.

**Comment by mananjadhav** [comment]
> make it so that all the avatars fit in the room header in all scenarios.

@puneetlath I didn't get you mean by this. Can you elaborate?

**Comment by abdulrahuman5196** [comment]
I assume this is the permutation @mananjadhav 

>On the OP it is mentioned "Once we get to 5, we use a smaller avatar size", then this should only apply to smallScreenWidth cases right? The UI permutation will be,
case 1 : isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE
case 2: isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.SMALL
case 3: not isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE
case 3: not isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.LARGE

as mentioned here - https://github.com/Expensify/App/issues/15929#issuecomment-1483766334

Correct me if i am wrong @puneetlath

**Comment by puneetlath** [comment]
>@puneetlath I didn't get you mean by this. Can you elaborate?

I just meant that no matter the screen size, we should make it so that all 7 avatars fit here. Given that we know that 7 is the max there will be. 

<img width="1111" alt="image" src="https://user-images.githubusercontent.com/1371996/235705655-61e0b691-2cd2-487d-94ac-0c2331f417b9.png">

**Comment by mananjadhav** [comment]
Cool. Thanks for clarifying. I meant for the smaller screen, it should auto adjust. Understood from the previous comments.

**Comment by abdulrahuman5196** [comment]
@puneetlath  @mananjadhav @roryabraham 

>I just meant that no matter the screen size, we should make it so that all 8 avatars fit here. Given that we know that 8 is the max there will be.

**How much smaller screens are we thinking to support?** There has to be a min supported width below which we won't support(If not we have to reduce the avatar size to be shown accordingly)

I am adding the permutations based on sizes I tested from the OP screenshot and their approx worst case width requirement for the MultipleAvatars component including all avatars to be rendered. Based on the permutations I provided during proposal

**Case 1:** isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE(80px one avatar) - 240px(approx) min width required(excluding paddings)
**Case 2:** isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.MEDIUM(52px one avatar) - 312px(approx) min width required
**Case 3:** not isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE - 240px(approx) min width required
**Case 4:** not isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.LARGE - 480px(approx) min width required

Note: If the screen size is below the required min width, all the avatars won't be shown and we might see avatars getting clipped.


From the calculation, I don't think case 3 and 4 will have any issue since its big screen with 800px >= width.
**Only case we have to think of 1 and 2. If we are expecting to support screen sizes less than 350px, we can reduce the avatar size from medium to small for case 2 or add another permutation**


If we could agree on the expected behaviour of how small we want to support I can provide better information. Mostly these are design decisions, so someone from design team should be able to help out as well if required.



**Reference:**

https://github.com/Expensify/App/blob/8c9e258f2543b6c454a42df82f259c2a8c73b1ed/src/styles/StyleUtils.js#L45-L55

https://github.com/Expensify/App/blob/8c9e258f2543b6c454a42df82f259c2a8c73b1ed/src/styles/variables.js#L31-L41

**Comment by abdulrahuman5196** [comment]
Gentle ping on the above. @puneetlath @roryabraham @mananjadhav 

It's blocking me to start on the implementation since it's a requirement confirmation.

**Comment by puneetlath** [comment]
@shawnborton do you have a sense of what the smallest screen size is that we should "reasonably" support?

**Comment by abdulrahuman5196** [comment]
@shawnborton 

>Can you catch me up on what the goal of this issue is? is it to always show all of the avatars of the group chat members, no matter the screen size?
If that's the case we might need to rethink of how this would work on mobile. For example, I'm not sure that we have the width to support 7 avatars on mobile, but we could try.

Yes. That's the gist of the issue. Issue creator can correct me if wrong. One correction is we need to support 8 avatars.

>In terms of minimum screen size, I would think maybe iPhone SE or something like that? Maybe we can make this responsive and if we have the available width, we can show avatars, and if not, we can fall back on something similar to what we currently have.


Even in the current behaviour it seems to break for iPhone SE first generation size of small phone 

![Simulator Screen Shot - Iphone SE test - 2023-05-08 at 19 09 44](https://user-images.githubusercontent.com/46707890/236846750-097bf284-3d3c-4ae9-8333-1e86ef750321.png)


I tried a quick code to show the expected behaviour
**For the new proposal.**
Case 1: isSmallScreenWidth and 4 avatars = AVATAR_SIZE.LARGE,
This seems to work fine


![Simulator Screen Shot - Iphone SE test - 2023-05-08 at 19 11 06](https://user-images.githubusercontent.com/46707890/236846800-3e045395-acba-43b8-a95a-a9404410d45a.png)

Case 2: isSmallScreenWidth and above 4 avatars = AVATAR_SIZE.MEDIUM(52px one avatar) 
Here AVATAR_SIZE.MEDIUM(52px one avatar) seems to be breaking so I choose AVATAR_SIZE.DEFAULT(40px) which is working fine.

![Simulator Screen Shot - Iphone SE test - 2023-05-08 at 19 07 36](https://user-images.githubusercontent.com/46707890/236847127-a60f4f9c-25a8-4739-a5dc-2940b54fdbaa.png)


Furthermore we can dynamically reduce sizes of avatar based on screen sizes to add to the above 4 cases but it might be ideal experience to have more sizes involved.




**I think if we want to consider Iphone SE 1st generation phone to be a min supported phone(which is not supported currently even), then we are good to go IMO.**

**Comment by shawnborton** [comment]
@puneetlath curious for your thoughts here since I believe this was your idea re: the empty state avatars. 

I think i would prefer to keep them large, and just show as many as we can based on screen width.

**Comment by puneetlath** [comment]
My thought was that given that we know the max number is 8, there's no reason why we can't always just show all 8. 

I personally think this (8 small ones): 
<img width="395" alt="image" src="https://github.com/Expensify/App/assets/1371996/74df346a-56f6-4076-8491-acb04388ef96">


Looks much better than this (4 big ones with +4): 
<img width="399" alt="image" src="https://github.com/Expensify/App/assets/1371996/d77b3637-49a1-44e3-8209-441d9d142145">

But I also don't feel super strongly about it if you don't agree @shawnborton.

**Comment by shawnborton** [comment]
It looks like your mockup still doesn't quite fit in this phone width though, so we'd need to make them even smaller... which I think ends up looking worse. 

We could potentially do some kind of stacked layout?
<img width="990" alt="image" src="https://github.com/Expensify/App/assets/2319350/d9620760-4008-458f-9193-d1229b5a281e">

**Comment by puneetlath** [comment]
I like that left, overlapping stacked one. That looks good to me.

**Comment by roryabraham** [comment]
It seems like we're still waiting on a finalized design here, but @shawnborton is OOO this week.

**Comment by puneetlath** [comment]
@shawnborton are you cool with going forward with the stacked approach on the left?

**Comment by puneetlath** [comment]
Of course 😄

**Comment by shawnborton** [comment]
Okay so I think I am not a fan of the diagonal stack. How about something like this?

<img width="1028" alt="image" src="https://github.com/Expensify/App/assets/2319350/9f37e322-db80-4eb0-9635-a8191e81327e">

**Comment by puneetlath** [comment]
Oh nice, I'm a fan!

**Comment by shawnborton** [comment]
I thought it felt more visually balanced and matches how the heading isn't quite as wide as the content below. So we basically create that same shape:
<img width="370" alt="image" src="https://github.com/Expensify/App/assets/2319350/dfa5fa45-66b0-47d1-9549-0876f0053e82">


Let me know if that makes sense.

**Comment by mananjadhav** [comment]
Now I see. Thanks for clarifying.

**Comment by mananjadhav** [comment]
@puneetlath @shawnborton What are the next steps? Opening up for proposals?

**Comment by shawnborton** [comment]
That makes sense to me

**Comment by puneetlath** [comment]
We had previously assigned @abdulrahuman5196. Do you still want to do it @abdulrahuman5196? If not, I agree that we should open it up.

**Comment by abdulrahuman5196** [comment]
Hi @puneetlath, I am fine to open it up for proposals. I am full at the moment. 

Note: Anyone going to work with MultipleAvatars to solve this issue, kindly note there is some refractor expected on that via https://github.com/Expensify/App/issues/19526

**Comment by roryabraham** [comment]
Going to double the price here to see if we get some fresh proposals.

**Comment by mananjadhav** [comment]
Thanks for the proposal @WikusKriek. I agree with most of the aspect of your proposal. Would you please elaborate on the following?

> A small function can handle the cutting of the icons into the appropriate length rows to match the design.

What kind of logic does the function include?

 > We can then map through each row and apply the same logic as horizontal stacking for each row.

What do we mean by apply the same logic?

----

@puneetlath @grgia do we have any ETA on [Refactor MultipleAvatars](https://github.com/Expensify/App/issues/19526). Do you folks think we can continue with this issue, while the refactor is planned?

**Comment by grgia** [comment]
@mananjadhav I don't think that [this refactor](https://github.com/Expensify/App/issues/19526) blocks this issue, you should be good to proceed. I can also prioritize that refactor if you'd prefer.

**Comment by mananjadhav** [comment]
I agree it won't block, was just concerned about splitting `MultipleAvatars` into two components could affect this one?

**Comment by mananjadhav** [comment]
I agree with @WikusKriek proposal [here](https://github.com/Expensify/App/issues/15929#issuecomment-1595869495). We can work out some of the implementation details in the PR.

🎀 👀 🎀 C+ Review.

**Comment by puneetlath** [comment]
I'll create a new job to invite you to. And yes, if we increase the scope, we'll definitely increase the price accordingly.

**Comment by roryabraham** [comment]
@puneetlath was more involved in the proposals and reviewed the PR while I was OOO, so I'm going to reassign this to him. Credit where credit's due.

**Comment by puneetlath** [comment]
Thanks for reminding me @WikusKriek. I added a second milestone to the contract to account for that.

**Comment by mananjadhav** [comment]
@puneetlath I've requested the payment via NewDot.

**Comment by anmurali** [comment]
@puneetlath can you confirm ^^ makes sense. I need to approve and pay this on New Dot

**Comment by puneetlath** [comment]
It should be $4k in total @anmurali. It was a $2k job and we doubled the scope, so it became $4k. No timeline bonus. 

I just paid out @WikusKriek that amount in Upwork so @mananjadhav just needs to be paid that as well.

**Comment by puneetlath** [comment]
It should be $4k in total @anmurali. It was a $2k job and we doubled the scope, so it became $4k. No timeline bonus. 

I just paid out @WikusKriek that amount in Upwork so @mananjadhav just needs to be paid that as well.

**Comment by anmurali** [comment]
Approved to Manan for $4k

**Comment by puneetlath** [comment]
Ok great. Closing this out then. Thanks everyone!
